### ISSUE TITLE
SNOW-563009: Issues connecting via proxy configuration settings with 1.6.8 client

### ISSUE THREAD

**Issue by denodaeus** [snowflakedb/snowflake-connector-nodejs#280]
1. What version of NodeJS are you using (`node --version` and `npm --version`)?
node v14.18.2, npm 6.14.15

3. What operating system and processor architecture are you using?
3.10.0-1160.49.1.el7.x86_64 (RHEL 7 Linux)
Red Hat Enterprise Linux Server release 7.9 (Maipo)

4. What are the component versions in the environment (`npm list`)?
Good (1.6.7):
```
─┬ snowflake-sdk@1.6.7
│ │ ├─┬ @azure/storage-blob@12.9.0
│ │ │ ├─┬ @azure/abort-controller@1.0.4
│ │ │ │ └── tslib@2.2.0 deduped
│ │ │ ├─┬ @azure/core-http@2.2.4
│ │ │ │ ├── @azure/abort-controller@1.0.4 deduped
│ │ │ │ ├── @azure/core-asynciterator-polyfill@1.0.2
│ │ │ │ ├─┬ @azure/core-auth@1.3.2
│ │ │ │ │ ├── @azure/abort-controller@1.0.4 deduped
│ │ │ │ │ └── tslib@2.2.0 deduped
│ │ │ │ ├── @azure/core-tracing@1.0.0-preview.13 deduped
│ │ │ │ ├── @azure/logger@1.0.3 deduped
│ │ │ │ ├─┬ @types/node-fetch@2.6.1
│ │ │ │ │ ├── @types/node@14.17.6 deduped
│ │ │ │ │ └─┬ form-data@3.0.1
│ │ │ │ │   ├── asynckit@0.4.0 deduped
│ │ │ │ │   ├── combined-stream@1.0.8 deduped
│ │ │ │ │   └── mime-types@2.1.30 deduped
│ │ │ │ ├─┬ @types/tunnel@0.0.3
│ │ │ │ │ └── @types/node@14.17.6 deduped
│ │ │ │ ├─┬ form-data@4.0.0
│ │ │ │ │ ├── asynckit@0.4.0 deduped
│ │ │ │ │ ├── combined-stream@1.0.8 deduped
│ │ │ │ │ └── mime-types@2.1.30 deduped
│ │ │ │ ├─┬ node-fetch@2.6.7
│ │ │ │ │ └─┬ whatwg-url@5.0.0
│ │ │ │ │   ├── tr46@0.0.3
│ │ │ │ │   └── webidl-conversions@3.0.1
│ │ │ │ ├── process@0.11.10 deduped
│ │ │ │ ├─┬ tough-cookie@4.0.0
│ │ │ │ │ ├── psl@1.8.0 deduped
│ │ │ │ │ ├── punycode@2.1.1
│ │ │ │ │ └── universalify@0.1.2
│ │ │ │ ├── tslib@2.2.0 deduped
│ │ │ │ ├── tunnel@0.0.6
│ │ │ │ ├── uuid@8.3.2
│ │ │ │ └── xml2js@0.4.23 deduped
│ │ │ ├─┬ @azure/core-lro@2.2.4
│ │ │ │ ├── @azure/abort-controller@1.0.4 deduped
│ │ │ │ ├── @azure/core-tracing@1.0.0-preview.13 deduped
│ │ │ │ ├── @azure/logger@1.0.3 deduped
│ │ │ │ └── tslib@2.2.0 deduped
│ │ │ ├─┬ @azure/core-paging@1.2.1
│ │ │ │ ├── @azure/core-asynciterator-polyfill@1.0.2 deduped
│ │ │ │ └── tslib@2.2.0 deduped
│ │ │ ├─┬ @azure/core-tracing@1.0.0-preview.13
│ │ │ │ ├── @opentelemetry/api@1.1.0
│ │ │ │ └── tslib@2.2.0 deduped
│ │ │ ├─┬ @azure/logger@1.0.3
│ │ │ │ └── tslib@2.2.0 deduped
│ │ │ ├── events@3.3.0
│ │ │ └── tslib@2.2.0 deduped
│ │ ├─┬ agent-base@4.3.0
│ │ │ └─┬ es6-promisify@5.0.0
```

Bad (1.6.8)
```
└─┬ snowflake-sdk@1.6.8
  ├─┬ @azure/storage-blob@12.9.0
  │ ├─┬ @azure/abort-controller@1.0.4
  │ │ └── tslib@2.3.1 deduped
  │ ├─┬ @azure/core-http@2.2.4
  │ │ ├── @azure/abort-controller@1.0.4 deduped
  │ │ ├── @azure/core-asynciterator-polyfill@1.0.2
  │ │ ├─┬ @azure/core-auth@1.3.2
  │ │ │ ├── @azure/abort-controller@1.0.4 deduped
  │ │ │ └── tslib@2.3.1 deduped
  │ │ ├── @azure/core-tracing@1.0.0-preview.13 deduped
  │ │ ├── @azure/logger@1.0.3 deduped
  │ │ ├─┬ @types/node-fetch@2.6.1
  │ │ │ ├── @types/node@17.0.22
  │ │ │ └─┬ form-data@3.0.1
  │ │ │   ├── asynckit@0.4.0 deduped
  │ │ │   ├── combined-stream@1.0.8 deduped
  │ │ │   └── mime-types@2.1.35 deduped
  │ │ ├─┬ @types/tunnel@0.0.3
  │ │ │ └── @types/node@17.0.22 deduped
  │ │ ├─┬ form-data@4.0.0
  │ │ │ ├── asynckit@0.4.0 deduped
  │ │ │ ├── combined-stream@1.0.8 deduped
  │ │ │ └── mime-types@2.1.35 deduped
  │ │ ├─┬ node-fetch@2.6.7
  │ │ │ └─┬ whatwg-url@5.0.0
  │ │ │   ├── tr46@0.0.3
  │ │ │   └── webidl-conversions@3.0.1
  │ │ ├── process@0.11.10
  │ │ ├─┬ tough-cookie@4.0.0
  │ │ │ ├── psl@1.8.0 deduped
  │ │ │ ├── punycode@2.1.1 deduped
  │ │ │ └── universalify@0.1.2
  │ │ ├── tslib@2.3.1 deduped
  │ │ ├── tunnel@0.0.6
  │ │ ├── uuid@8.3.2
  │ │ └─┬ xml2js@0.4.23
  │ │   ├── sax@1.2.4
  │ │   └── xmlbuilder@11.0.1
  │ ├─┬ @azure/core-lro@2.2.4
  │ │ ├── @azure/abort-controller@1.0.4 deduped
  │ │ ├── @azure/core-tracing@1.0.0-preview.13 deduped
  │ │ ├── @azure/logger@1.0.3 deduped
  │ │ └── tslib@2.3.1 deduped
  │ ├─┬ @azure/core-paging@1.2.1
  │ │ ├── @azure/core-asynciterator-polyfill@1.0.2 deduped
  │ │ └── tslib@2.3.1 deduped
  │ ├─┬ @azure/core-tracing@1.0.0-preview.13
  │ │ ├── @opentelemetry/api@1.1.0
  │ │ └── tslib@2.3.1 deduped
  │ ├─┬ @azure/logger@1.0.3
  │ │ └── tslib@2.3.1 deduped
  │ ├── events@3.3.0
  │ └── tslib@2.3.1
  ├─┬ agent-base@6.0.2
  │ └─┬ debug@4.3.4
  │   └── ms@2.1.2 deduped
```

6. What did you do?
a. Install the 1.6.8 version of the lib
b. configure proxyHost, proxyPort, and proxyProtocol
c. Observe error messages about connecting with the client, see below

8. What did you expect to see?
Similar connectivity to 1.6.7

10. What did you see instead?
The following errors in our ORM (sequelize):
```
2022-03-21T13:16:20: [Nest] 11568 - 03/21/2022, 1:16:20 PM [ExceptionsHandler] Network error. Could not reach Snowflake. +18442ms
2022-03-21T13:16:20: SequelizeConnectionError: Network error. Could not reach Snowflake.
```
And, in the client logs with TRACE enabled:
```
2022-03-21T13:16:20: {"level":"TRACE","message":"[1:16:20.1620 PM]: [https://<ACCOUNT>.privatelink.snowflakecomputing.com/session/v1/login-request?requestId=9d53fa0d-577c-4993-b8c9-54a781ea4926&databaseName=<DB>&schemaName=<SCHEMA>&roleName=<ROLE>&clientStartTime=1647882962575&retryCount=4"}]
2022-03-21T13:16:20: {"level":"DEBUG","message":"[1:16:20.1620 PM]: Failed to all retries to SF."}

```

And a stack trace related to agent-base:

```
2022-03-21T13:16:20: Error [NetworkError]: Network error. Could not reach Snowflake.
2022-03-21T13:16:20: at createError (/var/opt/ICE-Vantage-Bestex/node_modules/snowflake-sdk/lib/errors.js:535:15)
2022-03-21T13:16:20: at Object.exports.createNetworkError (/var/opt/ICE-Vantage-Bestex/node_modules/snowflake-sdk/lib/errors.js:260:10)
2022-03-21T13:16:20: at Object.callback (/var/opt/ICE-Vantage-Bestex/node_modules/snowflake-sdk/lib/services/sf.js:572:26)
2022-03-21T13:16:20: at Request.<anonymous> (/var/opt/ICE-Vantage-Bestex/node_modules/snowflake-sdk/lib/http/base.js:107:25)
2022-03-21T13:16:20: at Request._callback (/var/opt/ICE-Vantage-Bestex/node_modules/lodash/lodash.js:10118:25)
2022-03-21T13:16:20: at Request.requestRetryReply [as reply] (/var/opt/ICE-Vantage-Bestex/node_modules/requestretry/index.js:150:19)
2022-03-21T13:16:20: at Request.<anonymous> (/var/opt/ICE-Vantage-Bestex/node_modules/requestretry/index.js:191:10) {
2022-03-21T13:16:20: code: 401001,
2022-03-21T13:16:20: cause: Error: "agent-base" has no default implementation, you must subclass and override `callback()`
2022-03-21T13:16:20: at HttpsProxyAgent.callback (/var/opt/ICE-Vantage-Bestex/node_modules/agent-base/dist/src/index.js:73:19)
2022-03-21T13:16:20: at /var/opt/ICE-Vantage-Bestex/node_modules/agent-base/dist/src/promisify.js:6:16
2022-03-21T13:16:20: at new Promise (<anonymous>)
2022-03-21T13:16:20: at HttpsProxyAgent.promisifiedCallback (/var/opt/ICE-Vantage-Bestex/node_modules/agent-base/dist/src/promisify.js:5:16)
2022-03-21T13:16:20: at HttpsProxyAgent.addRequest (/var/opt/ICE-Vantage-Bestex/node_modules/agent-base/dist/src/index.js:184:38)
2022-03-21T13:16:20: at new ClientRequest (_http_client.js:296:16)
2022-03-21T13:16:20: at Object.request (https.js:314:10)
2022-03-21T13:16:20: at Request.start (/var/opt/ICE-Vantage-Bestex/node_modules/request/request.js:751:32)
2022-03-21T13:16:20: at Request.write (/var/opt/ICE-Vantage-Bestex/node_modules/request/request.js:1491:10)
2022-03-21T13:16:20: at end (/var/opt/ICE-Vantage-Bestex/node_modules/request/request.js:549:18) {
2022-03-21T13:16:20: attempts: 1
```

12. Add this to get standard output.
 
```
var snowflake = require('./snowflake');
snowflake.configure(
{
  logLevel: 'trace'
});
```

**Comment by sfc-gh-dszmolka** [comment]
hi @denodaeus  i might be little late to the party but see we released [proxy related fixes with 1.6.11 ](https://github.com/snowflakedb/snowflake-connector-nodejs/compare/v1.6.10...v1.6.11)
can you please confirm if the issue is indeed gone with 1.6.11 or later, or you still see it ? thank you!

**Comment by sfc-gh-dszmolka** [comment]
closing this for now - please reopen if it's still an issue with 1.6.11+
